### PR TITLE
Feature/proxy checkins

### PR DESCRIPTION
**Pull request by bonflintstone**
Allows users to add further checkins for people they are with

### CODE REVIEW

File: lib/actions/checkin.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { encrypt, toCSV } from '../crypto'
 interface Params {
   ticket: api.TicketReq
   guest: db.Guest
-  companyId: api.CompanyRes['id']
+  companyId?: api.CompanyRes['id']
```
Comment (timomeh) [inline lib/actions/checkin.ts:8]: Why is this optional now?

File: pages/my-checkins.tsx [hunk 2]
```diff
@@ -21,14 +22,33 @@ export default function MyCheckinsPage() {
...
     [setIsLoading, checkoutFn]
   )
 
+  // Sorts checkins by time and groups proxy checkins together with their "main" checkins
+  const groupedCheckins = React.useMemo(() => {
+    const sortedCheckins = checkinsInfo.data?.sort(
+      (c1, c2) => c1.enteredAt.getTime() - c2.enteredAt.getTime()
+    )
+
+    return sortedCheckins?.reduce((result: Checkin[][], checkin: Checkin) => {
```
Comment (timomeh) [inline pages/my-checkins.tsx:41]: Can you maybe add a small comment what's happening here? Took me a while to understand how  you're grouping them

File: ui/blocks/LastCheckins.tsx [hunk 3]
```diff
@@ -1,21 +1,70 @@
...
+
+        await checkinFn({ ticket, guest })
+
+        idRef.current = uuidv4()
+
+        queryCache.refetchQueries('checkins')
+
+        setShowProxyCheckin(false)
+
+        setLoading(false)
```
Comment (timomeh) [inline ui/blocks/LastCheckins.tsx:61]: Why all those blank lines after each line? 😅

**Review by timomeh — APPROVED** [review]
Just a few questions, nothing deal-breaking

File: lib/actions/checkin.ts (hunk 1 above)
Comment (bonflintstone) [reply lib/actions/checkin.ts:8]: So my thinking was that we do not save the proxy checkins, thus we do not need to add it to the guest in the db and if the companyId is `undefined` it just wont do anything.

File: pages/my-checkins.tsx (hunk 2 above)
Comment (bonflintstone) [reply pages/my-checkins.tsx:41]: Sure :+1:

File: ui/blocks/LastCheckins.tsx (hunk 3 above)
Comment (bonflintstone) [reply ui/blocks/LastCheckins.tsx:61]: Uhm...

**Pull request merged**